ethcore/src/executive.rs

```diff
@@ -261,17 +261,23 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 		}
 		trace!("Executive::call(params={:?}) self.env_info={:?}", params, self.info);
 
-		if self.engine.is_builtin(&params.code_address) {
-			// if destination is builtin, try to execute it
+		// if destination is builtin, try to execute it
+		if let Some(builtin) = self.engine.builtin(&params.code_address, self.info.number) {
+			// Engines aren't supposed to return builtins until activation, but
+			// prefer to fail rather than silently break consensus.
+			if !builtin.is_active(self.info.number) {
+				let msg = format!("Engine returned unactivated built-in at {}", params.code_address);
+				return Err(evm::Error::Internal(msg));
```

Comment:
Although, looking closer, this branch already indicates a consensus failure of some kind.